.github/workflows/main.yml

```diff
@@ -6,52 +6,64 @@ on:
       - main
 
 jobs:
+  validation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: gradle/wrapper-validation-action@v1
+
   checks:
     runs-on: macos-latest
     steps:
-
-      - name: Project Checkout
-        uses: actions/checkout@v2
-
-      - name: Setup JDK 1.8
-        uses: actions/setup-java@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
         with:
           java-version: 8
           distribution: 'zulu'
-
+      - uses: gradle/gradle-build-action@v2
       - name: Build project
         run: ./gradlew build --stacktrace
-
       - name: Assemble Test APK
         run: ./gradlew assembleAndroidTest --stacktrace
-
+      - name: AVD cache
+        uses: actions/cache@v3
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-16
```

Comment:
will need to figure out how to make this work with multiple devices (https://github.com/square/leakcanary/pull/2282)